cmd/app/add.go

```diff
@@ -85,38 +88,56 @@ func preRunAddCommand(ctx context.Context, clients *shared.ClientFactory) error
 	if !clients.Config.WithExperimentOn(experiment.BoltFrameworks) {
 		return nil
 	}
-	manifestSource, err := clients.Config.ProjectConfig.GetManifestSource(ctx)
-	if err != nil {
-		return err
-	}
-	if manifestSource.Equals(config.ManifestSourceRemote) {
-		return slackerror.New(slackerror.ErrAppInstall).
-			WithMessage("Apps cannot be installed due to project configurations").
-			WithRemediation(
-				"Install an app on app settings: %s\nLink an app to this project with %s\nList apps saved with this project using %s",
-				style.LinkText("https://api.slack.com/apps"),
-				style.Commandf("app link", false),
-				style.Commandf("app list", false),
-			).
-			WithDetails(slackerror.ErrorDetails{
-				slackerror.ErrorDetail{
-					Code:    slackerror.ErrProjectConfigManifestSource,
-					Message: "Cannot install apps with manifests sourced from app settings",
-				},
-			})
-	}
 	return nil
 }
 
 // RunAddCommand executes the workspace install command, prints output, and returns any errors.
 func RunAddCommand(ctx context.Context, clients *shared.ClientFactory, selection *prompts.SelectedApp, orgGrantWorkspaceID string) (context.Context, types.InstallState, types.App, error) {
 	if selection == nil {
-		selected, err := teamAppSelectPromptFunc(ctx, clients, prompts.ShowHostedOnly, prompts.ShowAllApps)
+		var isProductionApp bool
+		var err error
+
+		// When team flag is provided, default app environment to deployed if not specified.
+		// TODO(semver:major): Remove defaulting to deployed and require the environment flag to be set.
+		if clients.Config.TeamFlag != "" && addFlags.environmentFlag == "" {
```

Comment:
```suggestion
		if (clients.Config.TeamFlag != "" || clients.Config.AppFlag != "") && addFlags.environmentFlag == "" {
```

I'm not certain that this is the right logic, but we might want to use a default value if the `--app` flag is provided 🔍

Reply:
🗣️ This made me question if `--app` should accept `local` or `deployed` in future versions!

Preferring an `--environment` flag for that choice and leaving app IDs to `--app` might bring more straightforward logic here and in other selections, but this isn't a change we ought make now!

Reply:
Flag combinations can be confusing! 😝

I think we want the original combination of:

```golang
if clients.Config.TeamFlag != "" && addFlags.environmentFlag == "" {
```

* When `--team` is provided and `--environment` is not, then we default to `--environment deployed`.
* When `--app` is provided, we resolve the `--environment` from `apps.json` and `apps.dev.json`.
* When there is a mismatched combination, then the `AppSelect` will return an error that the app cannot be found.
  * Example: `--team T0123 --app A0123` where `A0123` doesn't belong to `T0123`

Reply:
🧠 question: I noticed the **warning** below hints at `--team` needing the `--environment` flag and was wondering if you think `--app` should also remove support for "local" or "deployed" in the next `semver:major`?

🧠 praise: Also, thanks for clearing up these default cases and flag setups alike 🤓

Reply:
> 🧠 question: I noticed the warning below hints at `--team` needing the `--environment` flag and was wondering if you think --app should also remove support for "local" or "deployed" in the next `semver:major`?

I won't lie, I thought `--app local` and `--app deployed` support was removed in v3.0.0 when we introduced `--environment`. This should definitely be removed in the next `semver:major` in favour of `--environment`. :two-cents:

In the meantime, we'll need to handle that use-case here, I suppose 🤔

Reply:
> 🤖 question: It's not so clear to me if an `install` can target different environments for a known app. This might not be an issue if so, but I'm wondering what you think.

An App ID can only be associated with one environment (1-to-1 relationship), so we don't want to allow the developer to override that. Right now, we error if the `--app` and `--environment` flags are used together. That seems the safest. A better approach is that we don't error if `--environment` matches the saved `--app`, but atm the error message asks you to remove the `--environment` flag.